src/components/Table/Internal/Body/Scroller.tsx

```diff
@@ -0,0 +1,211 @@
+import React, {
+  ForwardedRef,
+  useCallback,
+  useEffect,
+  useImperativeHandle,
+  useMemo,
+  useState,
+} from 'react';
+import { ButtonShape, ButtonSize, SecondaryButton } from '../../../Button';
+import { IconName } from '../../../Icon';
+import { ColumnType } from '../../Table.types';
+import { ScrollerProps, ScrollerRef } from '../OcTable.types';
+import { useDebounce } from '../../../../hooks/useDebounce';
+
+import styles from '../octable.module.scss';
+
+const BUTTON_HEIGHT = 36;
+
+export const Scroller = React.forwardRef(
+  <RecordType,>(
+    {
+      columns,
+      flattenColumns,
+      scrollBodyRef,
+      stickyOffsets,
+      scrollHeaderRef,
+    }: ScrollerProps<RecordType>,
+    ref: ForwardedRef<ScrollerRef>
+  ) => {
+    const [visible, setVisible] = useState<boolean>(false);
+    const [leftButtonVisible, setLeftButtonVisible] = useState<boolean>(false);
+    const [rightButtonVisible, setRightButtonVisible] = useState<boolean>(true);
+    const [buttonStyle, setButtonStyle] = useState<React.CSSProperties>({});
+
+    const scrollOffsets: number[] = useMemo(
+      () =>
+        columns.reduce(
+          (acc, column: ColumnType<RecordType>) => {
+            if (!column.fixed) {
+              acc.widths += column.width as number;
+              acc.columns.push(acc.widths);
+            }
+            return acc;
+          },
+          {
+            widths: 0,
+            columns: [0],
+          }
+        ).columns,
+      [columns]
+    );
+
+    const leftButtonLeft: number = useMemo(
+      () =>
+        stickyOffsets.left[flattenColumns.findIndex((column) => !column.fixed)],
+      [stickyOffsets, flattenColumns]
+    );
+
+    const rightButtonRight: number = useMemo(
+      () =>
+        stickyOffsets.right[
+          flattenColumns.findIndex((column) => column.fixed === 'right') - 1
+        ] ?? 0,
+      [stickyOffsets, flattenColumns]
+    );
+
+    const computePosition = useCallback(() => {
+      if (!scrollBodyRef.current) {
+        return;
+      }
+      const {
+        height: scrollBodyHeight,
+        top: scrollBodyTop,
+        bottom: scrollBodyBottom,
+      } = scrollBodyRef.current.getBoundingClientRect();
+      const { height: stickyHeaderHeight = 0 } =
+        scrollHeaderRef?.current?.getBoundingClientRect?.() || {};
+      const { height: viewportHeight } = document.body.getBoundingClientRect();
```

Comment:
Does this `getBoundingClientRect()` value update and hold on resize? If not, `offsetLeft` + `offsetWidth` calc may be needed.

Reply:
Not needed since we compute the positions on hover, if the viewport is resized the mouse will go out of the table anyway